Tests/Common/Util/ExtensionsTests.cs

```diff
@@ -165,6 +165,27 @@ public void ExchangeRoundDownInTimeZoneCorrectValuesAroundDaylightTimeChanges_Su
             Assert.AreEqual(expected, exchangeRounded);
         }
 
+        [Test]
+        // this unit test reproduces a fixed infinite loop situation, due to a daylight saving time change, GH issue 3707.
+        public void RoundDownInTimeZoneAroundDaylightTimeChanges()
+        {
+            // sydney time advanced Sunday, 6 October 2019, 02:00:00 clocks were turned forward 1 hour to
+            // Sunday, 6 October 2019, 03:00:00 local daylight time instead.
+            var timeAt = new DateTime(2019, 10, 6, 10, 0, 0);
+            var expected = new DateTime(2019, 10, 5, 10, 0, 0);
+
+            var exchangeRoundedAt = timeAt.RoundDownInTimeZone(Time.OneDay, TimeZones.Sydney, TimeZones.Utc);
+            // even though there is an entire 'roundingInterval' unit (1 day) between 'timeAt' and 'expected' round down
+            // is affected by daylight savings and rounds down the timeAt
+            Assert.AreEqual(expected, exchangeRoundedAt);
+
+            timeAt = new DateTime(2019, 10, 7, 10, 0, 0);
+            expected = new DateTime(2019, 10, 6, 11, 0, 0);
+
+            exchangeRoundedAt = timeAt.RoundDownInTimeZone(Time.OneDay, TimeZones.Sydney, TimeZones.Utc);
+            Assert.AreEqual(expected, exchangeRoundedAt);
+        }
```

Comment:
I think this test should test this case for all resolutions. In reviewing the code, there seemed to be an assumption of the FF resolution equaling the time zone shift. As I write this, I'm beginning to think that I might just be more accurately describing the case: the bug is limited to cases where the FF resolution equals the time zone shift. Is that accurate? Or does the bug arise in other cases as well?

Also, the new multi-line comment github feature is pretty sweet!

Reply:
The issue only happens for `Resolution.Daily/Time.OneDay` interval since that's the resolution which will round down hours. Not sure it won't happen in another combination of data and ff resolutions, what ends up mattering is the round down resolution as exposed in this unit test